File: templates/Features/BackgroundTask/Services/ActivationService_postaction.cs
﻿//{**
// These code blocks include the BackgroundTaskService Instance in the method `GetActivationHandlers()`
// and background task registration to the method `InitializeAsync()` in the ActivationService of your project.
//**}

using System;
//{[{
using Param_RootNamespace.Helpers;
//}]}

namespace Param_ItemNamespace.Services
{
    internal class ActivationService
    {
        private async Task InitializeAsync()
        {
            //{[{
            Singleton<BackgroundTaskService>.Instance.RegisterBackgroundTasks();
            //}]}
        }

        private IEnumerable<ActivationHandler> GetActivationHandlers()
        {
            //{[{
            yield return Singleton<BackgroundTaskService>.Instance;
            //}]}
//{--{
            yield break;//}--}
        }
    }
}

File: templates/Features/StoreNotificationsVB/Services/ActivationService_postaction.vb
﻿'{**
' These code blocks include the StoreNotificationsFeatureService Instance in the method `GetActivationHandlers()`
' and initializes it in the method `InitializeAsync()` in the ActivationService of your project.
'**}

'{[{
Imports Param_RootNamespace.Helpers
'}]}

Namespace Services
    Friend Class ActivationService
        Private Async Function InitializeAsync() As Task
            '{[{
            Await Singleton(Of StoreNotificationsFeatureService).Instance.InitializeAsync()
            '}]}
        End Function

        Private Iterator Function GetActivationHandlers() As IEnumerable(Of ActivationHandler)
            '{[{
            yield Singleton(Of StoreNotificationsFeatureService).Instance
            '}]}
'{--{
            yield Exit Function'}--}
        End Function
    End Class
End Namespace
